src/plugins.ts

```diff
@@ -231,11 +233,29 @@ export default class Plugins {
     this.isValidPlugin(c)
 
     if (install) {
-      await this.npm.install([], {
-        cwd: c.root,
-        logLevel: this.logLevel,
-        prod: false,
-      })
+      if (await fileExists(join(c.root, 'yarn.lock'))) {
+        this.debug('installing dependencies with yarn')
+        const yarn = new Yarn({config: this.config, logLevel: this.logLevel})
+        await yarn.install([], {
+          cwd: c.root,
+          logLevel: this.logLevel,
+        })
+      } else if (await fileExists(join(c.root, 'package-lock.json'))) {
```

Comment:
Would we want to also look for an `npm-shrinkwrap` file here?

Reply:
Possibly but `plugins link` is meant for local development so I think it's pretty safe to assume that there will be a `package-lock.json`

But I'll defer to your judgement here